app/routes/requirements.tsx

```diff
@@ -1,9 +1,117 @@
+import { Form, useLoaderData } from "@remix-run/react";
 import PageNav from "../components/PageNav";
+import { page } from "~/store/page.client";
+import { useState } from "react";
+import { v4 as uuidv4 } from "uuid";
+
+/** The client side loader, which runs after hydrate. Data from the serverLoader (`loader()`) is also available. */
+export async function clientLoader() {
+  return page.get();
+}
+clientLoader.hydrate = true;
+
+export function HydrateFallback() {
+  return <p>Loading...</p>;
+}
+
+const commonOptions = [
+  "Home",
+  "Mum's house",
+  "Dad's house",
+  "School",
+  "Daycare",
+  "Playground",
+  "Grandparents",
+  "Therapy",
+  "Church",
+  "Car",
+];
+
+const emptyMagnet = () => ({
+  id: uuidv4(),
+  name: "",
+  quantity: 1,
+});
 
 export default function Requirements() {
+  const pageData = useLoaderData<typeof clientLoader>();
+  const [magnets, setMagnets] = useState([...pageData.magnets, emptyMagnet()]);
+
+  function updateMagnet(newText: string, indexToUpdate: number) {
+    const newMagnets = magnets.map((magnet, index) => ({
+      ...magnet,
+      name: index === indexToUpdate ? newText : magnet.name,
+    }));
+    if (newMagnets[newMagnets.length - 1].name !== "") {
+      newMagnets.push(emptyMagnet());
+    }
+
+    setMagnets(newMagnets);
+    page.set({
+      ...pageData,
+      magnets: newMagnets.filter((m) => m.name !== ""),
+    });
+  }
+
+  function deleteMagnet(indexToDelete: number) {
+    console.log("magnets", magnets);
+    const newMagnets = magnets.filter(
+      (magnet, index) => index !== indexToDelete
+    );
+    if (newMagnets[newMagnets.length - 1].name !== "") {
+      newMagnets.push(emptyMagnet());
+    }
+
+    setMagnets(newMagnets);
+    page.set({
+      ...pageData,
+      magnets: newMagnets.filter((m) => m.name !== ""),
+    });
+    console.log("newMagnets", newMagnets);
+  }
+
+  function clickCommonOption(option: string) {
+    updateMagnet(option, magnets.length - 1);
+  }
+
+  const filteredCommonOptions = commonOptions.filter((option) => {
+    return !magnets
+      .map((m) => m.name.toLowerCase())
+      .includes(option.toLowerCase());
+  });
+
   return (
-    <div className="flex flex-col space-y-2">
-      <p>Add functionality to add list of requirements</p>
+    <div className="flex flex-col space-y-2 m-4">
+      <Form>
```

Comment:
Given we don't have an `action` or `clientAction` I'm not sure if this `<Form>` needs to be here or not